Add support for RSockets with Spring

dsyer opened the pull request:
Also generically can support structured events with any Spring
API that works with Encoder and Decoder. There's a sample for
the RSocket case with a simple request-response echo server.

slinkydeveloper commented on spring/src/main/java/io/cloudevents/spring/codec/CloudEventDecoder.java:
```diff
@@ -0,0 +1,73 @@
+/**
+ * Encoder for {@link CloudEvent CloudEvents}.
+ *
+ * @author Arjen Poutsma
+ * @since 5.0
```
Are these leftovers?

slinkydeveloper commented on spring/src/main/java/io/cloudevents/spring/codec/CloudEventEncoder.java:
```diff
@@ -0,0 +1,77 @@
+/**
+ * Encoder for {@link CloudEvent CloudEvents}.
+ *
+ * @author Arjen Poutsma
+ * @since 5.0
```
Are these leftovers?

slinkydeveloper commented on examples/spring-rsocket/src/test/java/io/cloudevents/examples/spring/DemoApplicationTests.java:
```diff
@@ -0,0 +1,59 @@
+						.withSource(URI.create("https://spring.io/foos")) //
+						.withType("io.spring.event.Foo") //
+						.withData(PojoCloudEventData.wrap(new Foo("Dave"),
+								foo -> mapper.writeValueAsBytes(foo)))
```
Use lambda reference

slinkydeveloper reviewed (commented):
Can you add some doc about these 2 new encoder/decoder here? https://github.com/cloudevents/sdk-java/blob/master/docs/spring.md

Can you also check all the copyrights and have them like this one? https://github.com/cloudevents/sdk-java/blob/master/core/src/main/java/io/cloudevents/core/CloudEventUtils.java#L1

dsyer commented:
I changed the copyrights to match the example (but also noticed that the other Spring module source code do not precisely match BTW). I also added a link and a mention to the docs - there isn't much detail there for any of the converters, but I'd be more than happy to add some notes independently of this PR.

slinkydeveloper commented:
> but also noticed that the other Spring module source code do not precisely match BTW

It would be nice if you can do another PR to fix them :smile: 

> I also added a link and a mention to the docs - there isn't much detail there for any of the converters, but I'd be more than happy to add some notes independently of this PR.

Sure, these docs are really drafted and i'll be more than happy to see them growing!
